doc/console/serial-console-HLD.md

```diff
@@ -0,0 +1,230 @@
+    - [6.2. ConfigDB schemas](#62-configdb-schemas)
+    - [6.3. CLI/YANG model Enhancements](#63-cliyang-model-enhancements)
+    - [6.4. Config DB Enhancements](#64-config-db-enhancements)
+    - [6.5. Manifest (if the feature is an Application Extension)](#65-manifest-if-the-feature-is-an-application-extension)
+- [7. Warmboot and Fastboot Design Impact](#7-warmboot-and-fastboot-design-impact)
+- [8. Restrictions/Limitations](#8-restrictionslimitations)
+- [9. Testing Requirements/Design](#9-testing-requirementsdesign)
+    - [9.1. Unit Test cases](#91-unit-test-cases)
+    - [9.2. System Test cases](#92-system-test-cases)
+- [10. Open/Action items - if any](#10-openaction-items---if-any)
+###  Revision  
+
+|  Rev  |   Date   |   Author   | Change Description |
+| :---: | :------: | :--------: | ------------------ |
+|  0.1  | 16/06/23 | Ivan Davydenko | Initial version    |
+
+##  1. Introduction  
+
+###  1.1. Scope  
+
+This hld doc for serial cli global configurations describes the requirements, architecture and general flow details of serial connection config in SONIC OS based switches.
+
+###  1.2. Definitions/Abbreviations 
+
+	serial - secure shell
+	TCP - Transmission Control protocol	
+	
+###  1.3. Overview 
+
+We want to allow configuring serial console global settings. This will feature will include two configurations on first phase.
+
+###  1.4. Requirements
+
+This feature requires a dedicated table in the configuration DB, enhancements of hostcfg demon and separate service that updates configutaion files. In order to update the serial-console configurations, we need to have write access in /etc/profile.d/, /proc/sys/kernel/sysrq and /etc/sysctl.d/95-sysrq-sysctl.conf.
+
+##  2. Architecture Design 
+###  2.1. Configuration modules
+![serial_console_config](serial_console_config.png)
+
+We want to enhance configDB to include table for serial-console global configurations. In addition, serial-config.service will be added to modify configuration files.
+
+##  3. High-Level Design 
+
+We want to enable serial-console configuration in SONIC. In order to do so will touch few areas in the system:
+1. configDB - to include a dedicated table for configurations
+2. hostcfg demon - to trigger dedicated service on config apply.
+3. OS config files - specific for this stage we are only /etc/profile.d/tmout-env.sh and /etc/sysctl.d/95-sysrq-sysctl.conf and /proc/sys/kernel/sysrq are going to be modifed by the serial-config.sh runned by serial-config.service .
+
+##### Flow diagram
+![serial_console_flow](serial_console_flow.png)
+### 3.1 Flow description
+When the feature is enabled, by modifying the DB manually, user will set serial-console configurations by modifing CONFIG_DB in SERIAL_CONSOLE table.
+
+The hostcfgd daemon will be extended to listen to confogurations from SERIAL_CONSOLE table and restarts the serial_console.service. Serial console script will read SERIAL_CONSOLE table and update config files accordingly.
```

Comment:
>serial_console.service [](http://example.com/codeflow?start=108&length=22)

Is it a vanilla Debian service, or part of this HLD? If latter, please explain what is its functionality?

Why single hostcfgd service is not enough?

Reply:
This is new service, not a vanilla Debian one. 

The rationale behind it creation is to update serial console config files (sysrq config, $TMOUT, and others that could be added in the future) before user allowed to start serial connection. 

We will achieve this by starting serial-console.service `Before=getty-pre.target`

Reply:
Is it possible to move its functionality into  hostcfgd service?

Reply:
I believe hostcfgd starts too late for this purpose, and it will impact the users config if the serial/ssh was started too early. Thats why I think in this case moving logic to hostcfgd is not a good solution.